pkg/provisioningv2/rke2/planner/encryptionkeyrotation.go

```diff
@@ -0,0 +1,440 @@
+	cp.Status.RotateEncryptionKeysPhase = phase
+	_, err := p.rkeControlPlanes.UpdateStatus(cp)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// encryptionKeyRotationPhaseFromPeriodic will attempt to extract the current phase (secrets-encrypt status) from the
+// plan by parsing the periodic output, or the one time output. In most cases, the periodic output will be the only one
+// populated, however the leader requires deterministic status when entering the apply sequence, as clusters with few
+// nodes can complete the 'prepare' sequence before the leader has time to rerun the periodic status command, which
+// could potentially cause a duplicate apply.
+func encryptionKeyRotationPhaseFromPeriodic(plan *planEntry) (string, error) {
+	output, ok := plan.Plan.PeriodicOutput[SecretsEncryptStatusCommand]
+	if !ok {
+		return "", ErrWaitingf("Could not extract current status from plan for [%s]: no output for status", plan.Machine.Name)
+	}
+	phase, err := encryptionKeyRotationPhaseFromOutput(plan, string(output.Stdout))
+	return phase, err
+}
+
+func encryptionKeyRotationPhaseFromOneTime(plan *planEntry) (string, error) {
+	output, ok := plan.Plan.Output[SecretsEncryptStatusCommand]
+	if !ok {
+		return "", ErrWaitingf("Could not extract current status from plan for [%s]: no output for status", plan.Machine.Name)
+	}
+	phase, err := encryptionKeyRotationPhaseFromOutput(plan, string(output))
+	return phase, err
+}
+
+func encryptionKeyRotationPhaseFromOutput(plan *planEntry, output string) (string, error) {
+	a := strings.Split(output, "\n")
+	if len(a) < 2 {
+		return "", ErrWaitingf("Could not extract current status from plan for [%s]: status output is incomplete", plan.Machine.Name)
+	}
+	a = strings.Split(a[1], ": ")
+	if len(a) < 2 {
+		return "", ErrWaitingf("Could not extract current status from plan for [%s]: status output is partially complete", plan.Machine.Name)
+	}
+	phase := a[1]
+	logrus.Debugf("Rotate encryption Status [%s]: %s", plan.Machine.Name, phase)
+	return phase, nil
+}
+
+func encryptionKeyRotationSecretsEncryptInstruction(cp *rkev1.RKEControlPlane, phase string) (plan.OneTimeInstruction, error) {
+	var command string
+
+	switch phase {
+	case EncryptionKeyRotationStart, EncryptionKeyRotationReencryptFinished:
+		command = EncryptionKeyRotationPrepare
+	case EncryptionKeyRotationPrepare:
+		command = EncryptionKeyRotationRotate
+	case EncryptionKeyRotationRotate:
+		command = EncryptionKeyRotationReencrypt
+	default:
+		return plan.OneTimeInstruction{}, fmt.Errorf("cannot create secrets-encrypt instruction for node with phase: [%s]", phase)
+	}
+
+	return plan.OneTimeInstruction{
+		Name:    SecretsEncryptApplyCommand,
+		Command: rke2.GetRuntimeCommand(cp.Spec.KubernetesVersion),
+		Args: []string{
+			"secrets-encrypt",
+			command,
+		},
+	}, nil
+}
+
+func encryptionKeyRotationStatusInstruction(cp *rkev1.RKEControlPlane) plan.OneTimeInstruction {
```

Comment:
I think `encryptionKeyRotationStatusOneTimeInstruction` is a more descriptive name.